users/tests/models_test.py

```diff
@@ -0,0 +1,83 @@
+from django.db import IntegrityError
+from django.test import TestCase
+
+
+from ..models import User
+
+
+user_data = {
+    'email': "john@doe.com",
+    'first_name': "John",
+    'last_name': "Doe",
+    'password': User.objects.make_random_password()
+}
+
+
+class UserTestCase(TestCase):
+    """ Testcase for user
+    """
+    def test_string_repr(self):
+        """ Check the string representation of the model
+        """
+        user = User.objects.create_user(**user_data)
+        self.assertEqual(user.__str__(), user.email)
+
+    def test_create_user(self):
+        """ Create user account and save
+        """
+        user = User.objects.create_user(**user_data)
+        self.assertIsInstance(user, User)
+        self.assertEqual(user.__str__(), user_data['email'])
+
+    def test_update_user(self):
+        """ Update the existing user and save
+        """
+        user = User.objects.create_user(**user_data)
+        self.assertIsInstance(user, User)
+
+        new_first_name = "Mark"
```

Comment:
no need to assign the first_name value to a variable if you are not going to reuse it.

directly assign the value to the field.
`user.first_name = "Mark"`

Reply:
Sir, the variable new_first_name = "Mark", I reuse it from the last line of code in this method.     
self.assertEqual(user.first_name, new_first_name) which check the user first_name.